dhis-2/dhis-services/dhis-service-administration/src/test/java/org/hisp/dhis/merge/orgunit/OrgUnitMergeServiceTest.java

```diff
@@ -0,0 +1,151 @@
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.merge.orgunit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import org.hisp.dhis.DhisSpringTest;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.dataset.DataSet;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.organisationunit.OrganisationUnitGroup;
+import org.hisp.dhis.period.MonthlyPeriodType;
+import org.hisp.dhis.period.PeriodService;
+import org.hisp.dhis.period.PeriodType;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.google.common.collect.Sets;
+
+/**
+ * @author Lars Helge Overland
+ */
+public class OrgUnitMergeServiceTest
+    extends DhisSpringTest
+{
+    @Autowired
+    private OrgUnitMergeService service;
+
+    @Autowired
+    private IdentifiableObjectManager idObjectManager;
+
+    @Autowired
+    private PeriodService periodService;
+
+    private PeriodType ptA;
+
+    private OrganisationUnit ouA;
+
+    private OrganisationUnit ouB;
+
+    private OrganisationUnit ouC;
+
+    @Override
+    public void setUpTest()
+    {
+        ptA = periodService.getPeriodTypeByClass( MonthlyPeriodType.class );
+
+        ouA = createOrganisationUnit( 'A' );
+        ouB = createOrganisationUnit( 'B' );
+        ouC = createOrganisationUnit( 'C' );
+
+        idObjectManager.save( ouA );
+        idObjectManager.save( ouB );
+        idObjectManager.save( ouC );
+    }
+
+    @Test
+    public void testGetFromQuery()
+    {
+        OrgUnitMergeQuery query = new OrgUnitMergeQuery();
+        query.setSources( Sets.newHashSet( BASE_OU_UID + 'A', BASE_OU_UID + 'B' ) );
+        query.setTarget( BASE_OU_UID + 'C' );
+
+        OrgUnitMergeRequest request = service.getFromQuery( query );
+
+        assertEquals( 2, request.getSources().size() );
+        assertTrue( request.getSources().contains( ouA ) );
+        assertTrue( request.getSources().contains( ouB ) );
```

Comment:
I created `org.hisp.dhis.utils.Assertions#assertContainsOnly` to avoid having to use multiple assertions for this type of set collection check.

Reply:
Okay that is useful.